src/tiledimage.js

```diff
@@ -2162,6 +2167,16 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
 
         const fallbackCompletion = getCompletionCallback();
 
+        if (!tileCacheCreated) {
```

Comment:
Hmm I might've found a solution, I realized I was allowing the manipulation pipeline to be called within two events, but properly managed only the invalidation (not tile load) one. If I instead fire tile-loaded every time and drop this way of thinking (that came from the fact that `tile-loaded` was historically THE event to modify data with), I can call incalidation logics only on new tiles (e.g. those that arrive and do not find a cache-equal ready tile). This also simplifies logics since invalidation will now require only one event, the code for handling the update pipeline is only on a single place, and the tile-loaded event can be fired for all tiles without a change. Though there will (usually) be 2 blocking events per new tile arrival, which slown down the tile load by a (probably insignificantly) small amount of time.

Reply:
That sounds like a good solution. I think we definitely still want to have the `tile-loaded` event, and simplifying the logic is good!